### DISCUSSION TITLE
Could you provide an example in typescript?

### DISCUSSION THREAD

**Discussion by jmcginty** [chimurai/http-proxy-middleware#746] [Q&A]
### Is this a feature request?

Yes

It would be great to have a simple example in typescript

Thanks !
James

**Comment by chimurai** [comment]
What kind of example are you looking for?
Just a hello world?

**Comment by jmcginty** [comment]
Yes, just something simple. 
This is what we have in javascript today: 

```
const proxy = require('http-proxy-middleware');

module.exports = function(app) {
  app.use(
    proxy('/api', {
      target: process.env.REACT_APP_PROXY,
      changeOrigin: true,
      secure: false
    })
  )
};
```

The equivalent in typescript would be enough for us.

**Comment by chimurai** [comment]
Would something like this help?

```typescript
import * as express from 'express';
import * as proxy from 'http-proxy-middleware';

const app = express();
app.use(proxy({
  target: 'http://jsonplaceholder.typicode.com',
  changeOrigin: true
}))

app.listen(3000);

// http://localhost:3000/users
```

**Comment by jmcginty** [comment]
Hi, this doesn't work for me, I am using create-react-app, and it does not use express. 

To create a new react app with typescript, do this: 

`npx create-react-app my-app --typescript`

(see) https://facebook.github.io/create-react-app/docs/adding-typescript

So, how would you add the proxy into a hello world app like this?

**Comment by chimurai** [comment]
I'm not a React user.

If I take your example and make it Typescript it would look like this:

```typescript
import * as proxy from 'http-proxy-middleware';

export default function(app) {
  app.use(
    proxy('/api', {
      target: process.env.REACT_APP_PROXY,
      changeOrigin: true,
      secure: false
    })
  )
};

```


Think it would be better to ask this question on their issue tracker.

Maybe this is something they are missing in their documentation...

**Comment by Naveenbc** [comment]
@jmcginty did you find a way to do it in react typescript?

**Comment by chimurai** [comment]
`setupProxy.ts` support belongs in the `create-react-app` project:

https://github.com/facebook/create-react-app/issues/8273
https://github.com/facebook/create-react-app/issues/6794
https://github.com/facebook/create-react-app/pull/10228
